lib/View.js

```diff
@@ -112,34 +152,244 @@ class View extends Node {
   }
 
   async observationCount ({ disableDistinct } = {}) {
-    const sources = new Set([...this.dimensions].reduce((all, d) => {
-      return all.concat(Array.isArray(d.source) ? d.source : [d.source])
-    }, []))
-
-    const source = [...sources][0]
+    const source = this.getMainSource()
 
     const { countQuery } = this.observationsQuery({ disableDistinct })
 
     const result = await source.client.query.select(countQuery, { operation: source.queryOperation })
 
+    if (!result.length) {
+      return NaN
+    }
+
     return parseInt(result[0].count.value)
   }
 
   static fromCube (cube) {
-    const cubeSource = CubeSource.fromSource(cube.source, cube)
+    if (!cube) {
+      throw Error('Needs a cube')
+    }
 
-    const view = new View({ parent: cube })
+    if (!cube.source) {
+      throw Error('Needs source')
+    }
 
+    cube.source.ptr
+      .addOut(ns.rdf.type, ns.view.CubeSource)
+      .addOut(ns.view.cube, toTerm(cube))
+
+    const view = new View({ parent: cube, source: cube.source })
     cube.dimensions.forEach(cubeDimension => {
-      view.addDimension(view.createDimension({
-        source: cubeSource,
-        path: cubeDimension.path,
-        as: cubeDimension.path
-      }))
+      const viewDimension = view.createDimension({
+        source: cube.source, path: cubeDimension.path, as: cubeDimension.path
+      })
+      view.addDimension(viewDimension)
     })
 
+    view.setDefaultColumns()
+
     return view
   }
+
+  setDefaultColumns () {
+    const orderingColumns = [...this.dimensions]
+    const sortAlphaNum = (a, b) => a.cubeDimensions[0].path.value.localeCompare(
+      b.cubeDimensions[0].path.value, 'en', { numeric: true })
+    orderingColumns.sort(sortAlphaNum)
+
+    const projectionPtr = this.ptr.out(ns.view.projection)
+    projectionPtr.addList(ns.view.columns,
+      orderingColumns.map(dimension => dimension.ptr.term))
+  }
+
+  orderBy (orderBy) {
```

Comment:
This method creates or gets the view's 'order by'. It works in the same way as 'offset' and 'limit' previously in the code.